src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et3ResponseService.java

```diff
@@ -207,55 +211,36 @@ private static String getRespondentNames(CaseData caseData) {
-     *      <ul>
-     *          <li>There is an issue retrieving the represented respondent indexes.</li>
-     *          <li>No represented respondent is found.</li>
-     *      </ul>
+     * @param userToken            the JWT bearer token representing the authenticated user
+     * @param caseData             the case data object to be updated with address information
+     * @param submissionReference  the reference string for identifying the current case submission
+     *
+     * @throws GenericServiceException if user details or organisation contact information
+     *                                  cannot be retrieved, or are missing or invalid
      */
-    public void validateAndExtractRepresentativeContact(String userToken, CaseData caseData, String submissionReference)
+    public void setRepresentativeMyHmctsContactAddress(String userToken, CaseData caseData, String submissionReference)
             throws GenericServiceException {
-        List<Integer> representedRespondentIndexes;
-        try {
-            representedRespondentIndexes = getRepresentedRespondentIndexes(userToken, submissionReference);
-        } catch (GenericServiceException gse) {
-            throw new GenericServiceException(gse.getMessage(),
-                    new Exception(gse),
-                    gse.getMessage(),
-                    submissionReference,
-                    "Et3ResponseService",
-                    "validateAndExtractRepresentativeContact");
-        }
-        if (CollectionUtils.isEmpty(representedRespondentIndexes) || representedRespondentIndexes.getFirst() == null) {
-            throw new GenericServiceException(ERROR_NO_REPRESENTED_RESPONDENT_FOUND,
-                    new Exception(ERROR_NO_REPRESENTED_RESPONDENT_FOUND),
-                    ERROR_NO_REPRESENTED_RESPONDENT_FOUND,
+
+        UserDetails userDetails = userIdamService.getUserDetails(userToken);
+        if (ObjectUtils.isEmpty(userDetails)) {
+            throw new GenericServiceException(ERROR_USER_NOT_FOUND,
+                    new Exception(ERROR_USER_NOT_FOUND),
+                    ERROR_USER_NOT_FOUND,
                     submissionReference,
                     "Et3ResponseService",
-                    "validateAndExtractRepresentativeContact");
+                    "setRepresentativeContactInfo - user not found");
         }
-        RepresentedTypeR representedTypeR = caseData.getRepCollection()
-                .get(representedRespondentIndexes.get(representedRespondentIndexes.getFirst())).getValue();
-        caseData.setEt3ResponsePhone(representedTypeR.getRepresentativePhoneNumber());
-        caseData.setEt3ResponseAddress(representedTypeR.getRepresentativeAddress());
+        OrganisationAddress organisationAddress = myHmctsService.getOrganisationAddress(userToken);
+        caseData.setEt3ResponseAddress(getAddress(organisationAddress));
+        caseData.setMyHmctsAddressText(getAddressAsText(organisationAddress));
```

Comment:
As you're using this in both in the mid event and the aboutToSubmit, you need to make sure when it's the aboutToSubmit callback, we are doing a caseData.setMyHmctsAddressText(null) so we aren't persisting this in the payload